packages/jest-haste-map/src/index.ts

```diff
@@ -629,17 +619,16 @@ class HasteMap extends EventEmitter {
   }
 
   private _buildHasteMap(data: {
-    deprecatedFiles: Array<{moduleName: string; path: string}>;
+    deprecatedFiles: FileData;
     hasteMap: InternalHasteMap;
   }): Promise<InternalHasteMap> {
     const {deprecatedFiles, hasteMap} = data;
     const map = new Map();
     const mocks = new Map();
     const promises = [];
 
-    for (let i = 0; i < deprecatedFiles.length; ++i) {
-      const file = deprecatedFiles[i];
-      this._recoverDuplicates(hasteMap, file.path, file.moduleName);
+    for (const [relativeFilePath, fileMetadata] of deprecatedFiles) {
```

Comment:
Isn't plain for loop faster than iterator for of?

Reply:
The `deprecatedFiles` data structure changed from an `Array` to a `Map` which is why I changed from a for loop. I don't think there is a more performant way to loop over a `Map`.

Regardless, this variable only contains files you just removed, so in most cases it won't contain many entries.

Reply:
Fair enough!